python/ccxt/async_support/base/exchange.py

```diff
@@ -480,6 +519,48 @@ def format_scientific_notation_ftx(self, n):
 
     # METHODS BELOW THIS LINE ARE TRANSPILED FROM JAVASCRIPT TO PYTHON AND PHP
 
+    def safe_prop(self, obj, key1, defaultValue=None):
+        return self.get_property(obj, key1, defaultValue)
+
+    def safe_prop_2(self, obj, key1, key2, defaultValue=None):
+        val1 = self.safe_prop(obj, key1)
+        if val1 is not None:
+            return val1
+        return self.safe_prop(obj, key2, defaultValue)
+
+    def set_exchange_prop_all_case(self, key: str, value: Optional[Any] = None):
+        setattr(self, key, value)
+        unCamelCasedKey = self.un_camel_case(key)
+        setattr(self, unCamelCasedKey, value)
+
+    def check_proxy_settings(self):
+        proxyUrl = self.safe_prop_2(self, 'proxyUrl', 'proxy_url')
```

Comment:
I think we're overcomplicating this check a bit, can't we do something like:

```
proxyUrl = this.proxyUrl ? this.proxyUrl : this.proxy_url
(...)
```

Reply:
oky

Reply:
question, using safeProp it would be much smaller code, otherwise, instead of :
```
 const proxyUrl = this.safeProp2 (this, 'proxyUrl', 'proxy_url');
```

We have to use:
```
let proxyUrl = undefined;
if (this.proxyUrl !== undefined) {
    proxyUrl = this.proxyUrl;
} else if (this.proxy_url !== undefined) {
    proxyUrl = this.proxy_url;
}
```

turning 1 line code into 6 lines. and it's not only one prop, we have to do that for 6-7 props, increaseing 6 lines into 42 lines of complex duplicate code.  so, i think it would be more complicated and uglier code

Reply:
@ttodua why 6 lines, as I suggested you can do it in one line as well `proxyUrl = this.proxyUrl ? this.proxyUrl : this.proxy_url`

Reply:
we would also make use of `safeProp` in other areas, as we talked before, we have some other instance properties, that are not being accessed with both cases (camelCase, snakeCase) in an unified manner, so we could extend this approach and benefit from it.

Reply:
because you cant have `proxyUrl = this.proxyUrl ? this.proxyUrl : this.proxy_url` in C# (or other similar langs) as it's not boolean

Reply:
@ttodua we have it extensively, and if you want to be really "correct" about falsy values sure, we can do this instead
`proxyUrl = this.proxyUrl !== undefined ? this.proxyUrl : this.proxy_url`

Reply:
but I'll add `!== undefined` and it would work, ok